### DISCUSSION TITLE
Changing form field options from required to not required.

### DISCUSSION THREAD

**Discussion by Thirty87** [bullhorn/career-portal#399] [Q&A]
I would like to change the form used for Applying for a job posting. The resume field is currently marked as required and we would like to mark it as not required. I can't seem to find the file(s) that I need to edit in order to apply the change to that field. We utilize the Wordpress Plugin Bullhorn-OSCP to embed the Career Portal into our site. Thanks in advance.

See attached image of the form for reference.

<img width="596" alt="Screen Shot 2021-06-28 at 4 06 48 PM" src="https://user-images.githubusercontent.com/86629656/123704568-64b97080-d82b-11eb-8905-27ea51e07c78.png">

**Comment by charlesabarnes** [comment]
Hello,

this is currently not an option to change in the Wordpress options.  This would require a code change to the core career portal code.  Fortunately, that code change is relatively simple.  The [required flag just need to be changed to false](https://github.com/bullhorn/career-portal/blob/master/src/app/apply-modal/apply-modal.component.ts#L155).  And when saving, change the post body to include something for the resume.  It could just include the first and last name since its a required field in the API.

**Reply by Thirty87** [reply]
Hey, thanks for the reply. I looked all over for the core files but can't seem to find them. I inherited this from a previous dev and not sure their method of how they stored their files. I've looked all over the web server and can't find the apply-modal.component.ts file. Any additional help would be appreciated on where I can look to make the edit.

**Reply by charlesabarnes** [reply]
Hey @Thirty87 the code used in the plugin is transpiled typescript code, so finding the specific javascript line in the bundled files is a difficult task.  You will need to clone this repository, then [build](https://github.com/bullhorn/career-portal#building) the code for use on your webserver

**Reply by jackhughes86** [reply]
Hey @charlesabarnes . Can you provide an example of how to change the post body to include the first name. I see the section of code you are talking about and tried to add firstName value. But not sure that I am doing it correctly.

**Comment by pbearne** [comment]
@Thirty87 if you are not able to sort this with OSCP came and talk to us at Matador Jobs. Our WordPress job broad for Bullhorn plugin replacement can be configured to do that with ease.
Paul@matadorjobs.com
PS we can do a lot more as well

**Comment by hackmasterS** [comment]
So it was easy enough to make the field not required, however the code still wants to do things with the resume (if even one is not uploaded). It wants to get the format (file extension) and also tries to send the file as well here "formData.append('resume', this.form.value.resume[0].file);"

We have made modifications to check for a file and have attempted to send fake data if no resume is uploaded but the API call responds with a 400 status code and this message {"errorMessage":"Unknown or badly structured command: /apply/17/raw.","errorMessageKey":"errors.badCommand"}

What data can we send to the API so that it allows an entry when a user does not wish to upload a resume?

**Reply by pbearne** [reply]
The problem with the amount of work you are doing with the core code in OSCP is that you will have to redo it each the OSCP is updated

Matador Jobs is set up to be happy with no CV

Paul

**Reply by hackmasterS** [reply]
Oh yes -- I am aware...it is just a few modifications to one single file though so was hoping someone could explain a work-around (although I know it to be a band-aid). I will have the client check out matador jobs too :)

**Reply by coLlenz** [reply]
Im getting this error after making the resume  not required, Bad File Uploaded: the request doesn't contain a multipart/form-data or multipart/mixed stream, content type header is application/json. Please help. Thaanks
